cfgov/jinja2/v1/_layouts/base.html

```diff
@@ -241,32 +241,42 @@
 {% block javascript scoped %}
 <!--[if gt IE 8]><!-->
 {# Include site-wide JavaScript. #}
-<script>
-    {% include '/js/routes/common.js' %}
-</script>
+<script src="{{ static('js/routes/common.js') }}"></script>
 
-{# Include template and page-specific JavaScript. #}
-{% macro page_template_js() %}
-    {# Check and include template-level JavaScript. #}
-    {% include '/js/routes/' + request.path.split('/')[1] + '/single.js' ignore missing %}
-
-    {# TODO: Remove Else conditional when this is fully moved to wagtail. #}
-    {# Check and include page-level JavaScript. #}
-    {% if page and page.media %}
-        {% for type, jsfiles in page.media.iteritems() %}
-            {% for js in jsfiles %}
-                {% include '/js/routes/on-demand/' + js ignore missing %}
-            {% endfor %}
-        {% endfor %}
-    {% else %}
-        {% include '/js/routes/' + request.path[1:] + 'index.js' ignore missing %}
-    {% endif %}
+{# Check and include template-level JavaScript. #}
+{% macro template_js() %}
+    {% include 'js/routes/' + request.path.split('/')[1] + '/single.js' ignore missing %}
 {% endmacro %}
-{% set js_source = page_template_js() | trim %}
+{% set js_source = template_js() | trim %}
 {% if js_source | length > 0 %}
-<script>
-    {{ js_source }}
-</script>
+<script src="{{ static('js/routes/' + request.path.split('/')[1] + '/single.js') }}"></script>
+{% endif %}
+
+{# Check and include component-level JavaScript. #}
+{% macro component_ondemand_js(js) %}
+    {% include 'js/routes/on-demand/' + js ignore missing %}
+{% endmacro %}
+
+{% if page and page.media %}
+   {% for type, jsfiles in page.media.iteritems() %}
+       {% for js in jsfiles %}
+           {% set js_source = component_ondemand_js(js) | trim %}
+           {% if js_source | length > 0 %}
+           <script src="{{ static('js/routes/on-demand/' + js) }}"></script>
```

Comment:
I'm not sure this is the wag to go because you potentially could have 10 scripts on a page. How would be feel about generating page-type ( browse, sublanding, etc...) specific bundles.

Reply:
Is there a way we could put this loop into a separate jinja template, which aggregates the scripts and inlines them into that template, and then `base.html` makes an external request to the result of that template. Basically, can jinja be used to dynamically generate a JS file?

Reply:
Separate page-type bundles would still be delivering a lot of unneeded code, right?

Reply:
> Is there a way we could put this loop into a separate jinja template, which aggregates the scripts and inlines them into that template, and then base.html makes an external request to the result of that template. Basically, can jinja be used to dynamically generate a JS file?

yes, you can use the compression [tags](https://github.com/cfpb/cfgov-refresh/pull/1409/files#diff-89b43ee86bbd4952dc44cf751cfd775cR205) and set the COMPRESS_JS_FILTERS to []. That seems hacky though, @rosskarchner probably knows a better way to do it.

Reply:
>> Separate page-type bundles would still be delivering a lot of unneeded code, right?
  
No, but you would need to generate every possible combination of components for the given page type.

Reply:
rather than lots of bundles that would be downloaded any time you visit a new page type, can we deliver the JS as one bundled file for now and investigate the impact of different optimization and delivery techniques? At least with one bundled file loaded asynchronously you have the ability to cache it for subsequent page views, reducing overall page load times for the user.

(I think this is what you're suggesting with this comment @ans)
> Is there a way we could put this loop into a separate jinja template, which aggregates the scripts and inlines them into that template, and then base.html makes an external request to the result of that template. Basically, can jinja be used to dynamically generate a JS file?